### PR TITLE
feat: enable access log or not

### PR DESCRIPTION
**Pull request by HelloBug0**
### What this PR does / why we need it:
satify the need of [#2713](https://github.com/apache/apisix/issues/2713)

### Pre-submission checklist:

* [X] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [X] Have you modified the corresponding document?
* [X] Is this PR backward compatible?

### CODE REVIEW

File: conf/config-default.yaml [hunk 1]
```diff
@@ -128,6 +128,7 @@ nginx_config:                     # config for render the template to genarate n
   #envs:                            # allow to get a list of environment variables
   #  - TEST_ENV
   http:
+    enable_access_log: true
```
Comment (tokers) [inline conf/config-default.yaml:131]: For the compatibilities' sake, the access_log should be enabled even if this field absent.

Comment (membphis) [reply conf/config-default.yaml:131]: agree

**Comment by membphis** [comment]
missing test case

File: conf/config-default.yaml (hunk 1 above)
Comment (HelloBug0) [reply conf/config-default.yaml:131]: OK, I will update this PR later.

**Comment by HelloBug0** [comment]
> missing test case

OK, I will add it. Would you like to tell me that is it OK if I add a new test file in the directory https://github.com/apache/apisix/tree/master/t/config-center-yaml? Or should I add some test cases in an existing file?

**Comment by spacewander** [comment]
@HelloBug0 
You can add it in https://github.com/apache/apisix/blob/master/.travis/apisix_cli_test.sh

File: .travis/apisix_cli_test.sh [hunk 2]
```diff
@@ -391,6 +391,52 @@ fi
 
 git checkout conf/config.yaml
```
Comment (spacewander) [inline .travis/apisix_cli_test.sh:392]: We don't need to checkout in the middle of a group of test, as we will change it later.

File: .travis/apisix_cli_test.sh [hunk 3]
```diff
@@ -391,6 +391,52 @@ fi
...
+    exit 1
+fi
+
+grep "access_log off;" conf/nginx.conf > /dev/null
+if [ $? -eq 0 ]; then
+    echo "failed: nginx.conf file find access_log off; when enable access log"
+    exit 1
+fi
+
+git checkout conf/config.yaml
```
Comment (spacewander) [inline .travis/apisix_cli_test.sh:415]: ditto

**Comment by membphis** [comment]
@HelloBug0 CI failed, please take a look at the output of CI

File: .travis/apisix_cli_test.sh [hunk 4]
```diff
@@ -389,12 +389,86 @@ if [ ! $? -eq 0 ]; then
...
+echo '
+nginx_config:
+  http:
+    enable_access_log: false
+    access_log_format: "$remote_addr - $remote_user [$time_local] $http_host test_enable_access_log"
+' > conf/config.yaml
+
+make init
+
+grep "test_enable_access_log" conf/nginx.conf > /dev/null
```
Comment (spacewander) [inline .travis/apisix_cli_test.sh:440]: Use `grep -c xx || true` for this line?

Comment (HelloBug0) [reply .travis/apisix_cli_test.sh:440]: Do you know the reason why this test case is failed? It is a little bit confusing~

Comment (spacewander) [reply .travis/apisix_cli_test.sh]: Because there is `set -e` at the top of the file.

**Comment by HelloBug0** [comment]
Finally, all checks have passed, whoopee~

File: .travis/apisix_cli_test.sh (hunk 4 above)
Comment (HelloBug0) [reply .travis/apisix_cli_test.sh]: Oh, I see. Thank you:smile:

**Review by membphis — APPROVED** [review]
LGTM

**Review by spacewander — APPROVED** [review]

File: .travis/apisix_cli_test.sh [hunk 5]
```diff
@@ -389,12 +389,86 @@ if [ ! $? -eq 0 ]; then
...
+    exit 1
+fi
+
+make run
+sleep 0.1
+curl http://127.0.0.1:9080/hi
+sleep 4
+tail -n 1 logs/access.log > output.log
+
+count_grep=`grep -c "test_enable_access_log_false" output.log || true`
```
Comment (tokers) [inline .travis/apisix_cli_test.sh:458]: We should remove the output.log after testing.

**Pull request merged**